addons/io_scene_gltf2/io/exp/gltf2_io_user_extensions.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2019 The glTF-Blender-IO authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import glob
+import imp
+import importlib
+import os.path
+import bpy
+from io_scene_gltf2.io.com.gltf2_io_extensions import Extension
+
+class GLTF2ExtensionMesh:
+    def __init__(self, blender_mesh, vertex_groups, modifiers):
+        self.blender_mesh = blender_mesh,
+        self.vertex_groups = vertex_groups,
+        self.modifiers = modifiers
+
+class GLTF2ExtensionNode:
+    def __init__(self, blender_object):
+        self.blender_object = blender_object 
+
+class GLTF2ExtensionAnimationChannelTarget:
+    def __init__(self, channels, blender_object, bake_bone):
+        self.channels = channels,
+        self.blender_object = blender_object,
+        self.bake_bone = bake_bone
+
+class GLTF2ExtensionAnimationChannels:
+    def __init__(self, channels, blender_object, bake_bone):
+        self.channels = channels,
+        self.blender_object = blender_object,
+        self.bake_bone = bake_bone
+
+class GLTF2ExtensionAnimationSamplers:
+    def __init__(self, channels, blender_object_if_armature, bake_bone, bake_channel):
+        self.channels = channels,
+        self.blender_object_if_armature = blender_object_if_armature,
+        self.bake_bone = bake_bone
+        self.bake_channel = bake_channel
+
+class GLTF2ExtensionAnimations:
+    def __init__(self, blender_action, blender_object):
```

Comment:
How does this work? What action/object is it getting? Because generally an animation will correspond to a collection of Blender actions.

Reply:
It works exactly the same way as in other exported element. The `__gather_extensions` in `gltf2_blender_gather_animations` receives a `blender_action` and a `blender_object`. I simply redirect those to the extension exporters.

Reply:
Yeah, I see that you just use whatever the `__gather_extensions` function take, but is that okay? Most of these are just returning None right now anyway so there's no guarantee their arguments are actually useful or sensible enough to become a public API.

For example, I just tested this and this gets called once for each (action,blender_object) in the animation, but only the value returned by the first call for each animation actually makes it into the `extensions` dict; the rest get dropped on the floor. I don't think that's how it should work.

Reply:
Well, if the __gather_extensions should receive another thing, the it's obvious it will be incorrect, but I'm not sure if its on the scope of this PR to fix what the __gather_extensions function should receive. 
If you think its on the scope, we can try to fix it, but probably we will have to review all the other __gather_extensions methods.